### PR TITLE
Fix the calculation of renames in the delta tree adapter.

### PR DESCRIPTION
**Pull request by maximAtanasov**
The delta tree endpoint does not properly recognize all renames. 
Currently a rename is recognized only if a file was renamed without any changes to it since it was added.
This could partly be fixed by setting the `objectHash` attribute on the relationship between a file and a commit instead of on the file node. However, that still woudn't be perfect as sometimes file contents change between renames. Internally git uses algorithms and has specific thresholds for detecting how much of a file has changed and then deciding if the change can still be labeled a rename. In this PR I just use JGit to get all renames between two arbitrary commits. This simplifies the adapter and also makes the query faster.

### CODE REVIEW

File: coderadar-core/src/main/java/io/reflectoring/coderadar/vcs/port/driven/GetRawCommitContentPort.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import io.reflectoring.coderadar.vcs.UnableToGetCommitContentException;
 import java.util.HashMap;
 import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
```
Comment (KilianKrause) [inline coderadar-core/src/main/java/io/reflectoring/coderadar/vcs/port/driven/GetRawCommitContentPort.java:6]: We could use the `Pair` from spring data and remove the `commons-lang3` dependency.

File: coderadar-core/build.gradle [hunk 2]
```diff
@@ -16,6 +16,7 @@ dependencies {
     compile "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.9.8"
     compile group: 'org.reflections', name: 'reflections', version: '0.9.11'
     testCompile "org.springframework.boot:spring-boot-starter-test:${version_spring_boot}"
+    compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.11'
```
Comment (KilianKrause) [inline coderadar-core/build.gradle:19]: This dependency can be removed if we decide to use the `Pair` class from spring data.

File: coderadar-vcs/src/main/java/io/reflectoring/coderadar/vcs/adapter/GetRawCommitContentAdapter.java [hunk 3]
```diff
@@ -36,13 +38,47 @@
     }
   }
 
+  public List<Pair<String, String>> getRenamesBetweenCommits(
```
Comment (KilianKrause) [inline coderadar-vcs/src/main/java/io/reflectoring/coderadar/vcs/adapter/GetRawCommitContentAdapter.java:41]: Missing `@Override` annotation.

Comment (maximAtanasov) [reply coderadar-vcs/src/main/java/io/reflectoring/coderadar/vcs/adapter/GetRawCommitContentAdapter.java:41]: Done

**Pull request merged**